### PR TITLE
Set the v2 and wrapped in window.q handler for issue #1248

### PR DESCRIPTION
**Pull request by linkel**
For #1248 

## Description:
- [x] Set v2 on the page object passed to any templates (e.g. 8d79b57)
- [x] Confirmed #test-body-mobile is present in the HTML
- [x] Loaded the page and view source. Identify all script tags in the page
- [x] Ensured the inline scripts in the page are wrapped in a window.q handler
- [x] Tested the page - is everything working as before?
- [x] Opened the page in a mobile browser 

I'm pretty new, so please let me know if there were other tests I should be performed before submitting this pull request. It looks like everything works as before.

openlib mobile view create acct

### CODE REVIEW

File: openlibrary/plugins/upstream/account.py [hunk 1]
```diff
@@ -225,20 +227,27 @@ def is_plugin_enabled(self, name):
     def POST(self):
         i = web.input('email', 'password', 'username', agreement="no")
         i.displayname = i.get('displayname') or i.username
-
+
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/account.py]: can we please remove this added whitespace? :)

File: openlibrary/plugins/upstream/account.py [hunk 2]
```diff
@@ -225,20 +227,27 @@ def is_plugin_enabled(self, name):
     def POST(self):
         i = web.input('email', 'password', 'username', agreement="no")
         i.displayname = i.get('displayname') or i.username
-
+        
         f = self.get_form()
+
```
Comment (mekarpeles) [inline openlibrary/plugins/upstream/account.py]: Suggestion: we might declare a variable `page = None` before checking e.g. validation etc and then at the bottom (before returning) set `page.v2 = True` once? (DRY: don't repeat ourselves)

Comment (jdlrobson) [reply openlibrary/plugins/upstream/account.py]: The CI problem was an issue on master which has since been fixed.

I've restarted the build. You may need to rebase it against upstream.

File: openlibrary/plugins/upstream/account.py [hunk 3]
```diff
@@ -225,20 +227,26 @@ def is_plugin_enabled(self, name):
     def POST(self):
         i = web.input('email', 'password', 'username', agreement="no")
         i.displayname = i.get('displayname') or i.username
-
```
Comment (jdlrobson) [inline openlibrary/plugins/upstream/account.py]: There's some unrelated whitespace changes here.

File: openlibrary/templates/account/create.html [hunk 4]
```diff
@@ -12,8 +12,10 @@ <h1>$_("Sign Up")</h1>
 <div id="contentBody">
 
 <script type="text/javascript">
-\$().ready(validateForms);
-\$().ready(function(){
+window.q.push( function() {
+    validateForms();
+});
+window.q.push( function(){
```
Comment (jdlrobson) [inline openlibrary/templates/account/create.html]: You can merge these into one single function push.
There's no reason to have more than one window.q.push call.

The window.q simply registers a callback which will run in a FIFO fashion when JS is loaded.

**Comment by mekarpeles** [comment]
re: @jdlrobson 
>Given that issue now impacts both login and sign up pages, it probably makes sense to fix these separately in another patch that benefits both these changes (cc @tabshaikh who's working on the login form.)

that sounds reasonable to me, this seems like a good candidate for a `good first issue`

> I'd like to see these buttons and inputs styled on mobile by explicitly adding some styles for ol.form input's and adding "btn primary" to the sign up and login button as this is how the form is showing me on an emulator [...]

:+1:

**Comment by linkel** [comment]
Thanks for the reviewing. I think I've addressed all the minor changes. What are the next steps? 

How do I know what whitespace is considered appropriate to keep and which whitespace is unnecessary? Is there a formatting guide for this?

File: openlibrary/plugins/upstream/account.py [hunk 5]
```diff
@@ -258,9 +256,9 @@ def POST(self):
         except ValueError as e:
             f.note = LOGIN_ERRORS['max_retries_exceeded']
             page = render['account/create'](f)
-            page.v2 = True
             return page
-
+
```
Comment (jdlrobson) [inline openlibrary/plugins/upstream/account.py]: redundant whitespace here.
Apart from this, I think this patch is good to go.

File: openlibrary/plugins/upstream/account.py [hunk 6]
```diff
@@ -225,20 +227,24 @@ def is_plugin_enabled(self, name):
     def POST(self):
         i = web.input('email', 'password', 'username', agreement="no")
         i.displayname = i.get('displayname') or i.username
-
```
Comment (jdlrobson) [inline openlibrary/plugins/upstream/account.py]: This line is still being unnecessarily removed.

Comment (LeadSongDog) [reply openlibrary/plugins/upstream/account.py]: @jdlrobson Are you concerned about the removal of a blank line 228? How is that a problem?

**Comment by mekarpeles** [comment]
These changes are looking much better to me. @jdlrobson any blockers for ?

**Comment by jdlrobson** [comment]
Please merge!!

**Comment by mekarpeles** [comment]
Congrats @linkel! Great work! Looking forward to seeing more of your contributions in the OL community ^.^
